code/linkfield/Blocks/LinksListBlock.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace SilverStripe\FrameworkTest\LinkField\Blocks;
+
+use DNADesign\Elemental\Models\BaseElement;
+use SilverStripe\LinkField\Form\LinkField;
+use SilverStripe\LinkField\Form\MultiLinkField;
+use SilverStripe\LinkField\Models\Link;
+use SilverStripe\LinkField\Models\EmailLink;
+use SilverStripe\LinkField\Models\PhoneLink;
+use SilverStripe\LinkField\Models\SiteTreeLink;
+
+class LinksListBlock extends BaseElement
+{
+    private static string $icon = 'font-icon-block-file-list';
+
+    private static string $table_name = 'LinksListBlock';
+
+    private static string $singular_name = 'Links List Block';
+
+    private static string $plural_name = 'Links List Blocks';
+
+    private static string $description = 'Displays a set of links which can be internal page links, external urls or '
+                                    . 'file links.';
+
+    private static bool $inline_editable = true;
+
+    private static array $has_one = [
+        'OneLink' => Link::class,
+    ];
+
+    private static $has_many = [
+        'ManyLinks' => Link::class . '.Owner',
+    ];
+
+    private static array $owns = [
+        'OneLink',
+        'ManyLinks',
+    ];
+
+    private static array $cascade_deletes = [
+        'OneLink',
+        'ManyLinks',
+    ];
+
+    private static array $cascade_duplicates = [
+        'OneLink',
+        'ManyLinks',
+    ];
+
+    public function getType(): string
+    {
+        return _t(self::class . '.BlockType', 'Links List');
+    }
+
+    public function getCMSFields()
+    {
+        $fields = parent::getCMSFields();
+
+        $fields->removeByName(['OneLinkID', 'ManyLinks']);
+
+        $fields->addFieldsToTab(
+            'Root.Main',
+            [
+                LinkField::create('OneLink')
+                ->setAllowedTypes([
+                    SiteTreeLink::class,
+                    EmailLink::class,
+                    PhoneLink::class
+                ]),
+                MultiLinkField::create('ManyLinks'),
+            ],
+        );
+
+        return $fields;
+    }
+}
```

Comment:
Updated. Unfortunately, we cannot use `TestOnly`. Please see [this comment](https://github.com/silverstripe/silverstripe-frameworktest/pull/162#issuecomment-1926449690). I created [new issue](https://github.com/silverstripe/silverstripe-behat-extension/issues/260) for this.

Reply:
@emteknetnz, for using `Extension` in these test classes I need to rebuild db, since I need to add new relations between classes. For doing this I need to update this [block of code](https://github.com/silverstripe/silverstripe-behat-extension/blob/99fd1ac53b8c98b5f5d96ab323219c7586a6e91b/src/Context/FixtureContext.php#L701-L705), that can be potential problem for another behat tests in another modules. Do you want me to update `FixtureContext` as part of this PR? Or if we are planning to get rid of `DataExtension` in the future then we will make required updates and all test classes will extend `DataExtension` for now? 
Note: As temporary solution I can enforce execution of `dev/build` as  additional step. What would you prefer?

Reply:
Don't worry about refactoring other things right now. For now simply make this a regular Extension and manually run dev/build